frontend/src/metabase/query_builder/components/VisualizationError/VisualizationError.tsx

```diff
@@ -0,0 +1,129 @@
+import { t } from "ttag";
+import { getIn } from "icepick";
+import cx from "classnames";
+
+import MetabaseSettings from "metabase/lib/settings";
+import { getEngineNativeType } from "metabase/lib/engine";
+import { ErrorMessage } from "metabase/components/ErrorMessage";
+import ErrorDetails from "metabase/components/ErrorDetails/ErrorDetails";
+import { isNotNull } from "metabase/lib/types";
+import { getShowMetabaseLinks } from "metabase/selectors/whitelabel";
+import { useSelector } from "metabase/lib/redux";
+import type Question from "metabase-lib/Question";
+import { VISUALIZATION_SLOW_TIMEOUT } from "../../constants";
+import {
+  QueryError,
+  QueryErrorHeader,
+  QueryErrorIcon,
+  QueryErrorTitle,
+  QueryErrorLink,
+  QueryErrorMessage,
+  QueryErrorContent,
+} from "./VisualizationError.styled";
+import { adjustPositions, stripRemarks } from "./utils";
+
+function EmailAdmin(): JSX.Element | null {
+  const hasAdminEmail = isNotNull(MetabaseSettings.adminEmail());
+  return hasAdminEmail ? (
```

Comment:
In some instances, I don't want to return `boolean` because it messes with the expected type for some props e.g. some props expect that the return type of this component would be `ReactElement` or ` JSX.Element` but `boolean` is incompatible.